src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -1457,6 +1494,1629 @@ class Test : System.Attribute
         }
 
         [Fact]
+        public void Scope_Catch_01()
+        {
+            var source =
+@"
+public class X
+{
+    public static void Main()
+    {
+    }
+
+    bool Dummy(params object[] x) {return true;}
+
+    void Test1()
+    {
+        try {}
+        catch when (TakeOutParam(out var x1) && x1 > 0)
```

Comment:
The tests as written are extremely difficult to review because of the large number of mixed positive and negative tests aggregated together. I cannot easily see both the test and the verification code on the same screen.